### PR TITLE
lmod: fix use of custom separator in prepend_path etc.

### PR DESCRIPTION
**Pull request by SteVwonder**
Fixes: #8736

### CODE REVIEW

**Comment by SteVwonder** [comment]
@alalazo: the travis tests on this PR are now passing.  This is now ready for your (or anyone else's) review.

File: lib/spack/spack/test/modules/lmod.py [hunk 1]
```diff
@@ -151,6 +152,19 @@ def test_alter_environment(self, modulefile_content, module_configuration):
...
 
+    def test_prepend_path_separator(self, modulefile_content,
+                                    module_configuration):
+        """Tests modifications to run-time environment."""
+
+        module_configuration('module_path_separator')
+        content = modulefile_content('module-path-separator')
+
+        for line in content:
+            if re.match(r'[a-z]+_path\("COLON"', line) is not None:
```
Comment (alalazo) [inline lib/spack/spack/test/modules/lmod.py]: Minor comment on style (not sure if everybody in the project here will agree, so feel free to discard this if you don't like it). This and the next condition can be written:
```python
if re.match(r'[a-z]+_path\("COLON"', line):
    ....
```
without the additional `is not None`. I find this style more readable.

File: share/spack/templates/modules/modulefile.lua [hunk 2]
```diff
@@ -72,6 +72,21 @@ end
 
 {% block environment %}
 {% for command_name, cmd in environment_modifications %}
+{% if cmd.separator != ':' %}
```
[the remark this answers is not in the document]
Comment (SteVwonder) [reply share/spack/templates/modules/modulefile.lua]: Sure! That is a much cleaner way of handling it.  If it works, would you like me to propagate this change to the tcl module file as well (https://github.com/spack/spack/blob/develop/share/spack/templates/modules/modulefile.tcl#L49)?  That is originally where I copied this control flow from.

[the remark this answers is not in the document]
Comment (SteVwonder) [reply share/spack/templates/modules/modulefile.lua]: Reporting back that this does in fact work.  :tada:

**Comment by alalazo** [comment]
@SteVwonder Thanks!

**Pull request merged**